src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1074,11 +1079,70 @@ def apply_fixes(self, fixes):
                 # Pass through any additional kwargs
                 **{k: getattr(self, k) for k in self.additional_kwargs},
             )
+            if fixes_applied:
+                self._validate_segment_after_fixes(dialect, fixes_applied, r)
             # Return the new segment with any unused fixes.
             return r, fixes
         else:
             return self, fixes
 
+    def _validate_segment_after_fixes(self, dialect, fixes_applied, segment):
+        """Checks correctness of new segment against match or parse grammar."""
+        found_error = False
+        root_parse_context = RootParseContext(dialect=dialect)
+        with root_parse_context as parse_context:
+            if getattr(segment, "match_grammar", None):
+                try:
+                    for seg in segment.segments:
+                        seg.pos_marker = replace(
+                            seg.pos_marker,
+                            templated_file=self.pos_marker.templated_file,
+                        )
+                    match_result = segment.match(
+                        deepcopy(
+                            tuple([seg for seg in segment.segments if not seg.is_meta])
+                        ),
+                        parse_context,
+                    )
```

Comment:
Would it be possible insert / replace new segments as a special type/Class.

`FixRawSegement(raw="my_whole.new_unlexed AND unparsed_segement")`

And also run a quick Lex on these small segments? around here. 
Giving us a simpler and more correct API for edits.

Reply:
@OTooleMichael: That's definitely something we can look at. Can you create a GitHub enhancement issue for this?